pandas/io/pytables.py

```diff
@@ -442,8 +442,8 @@ def __init__(self, path, mode=None, complevel=None, complib=None,
             mode = 'a'
         self._mode = mode
         self._handle = None
-        self._complevel = complevel
-        self._complib = complib
+        self._complevel = complevel if complevel else 0
```

Comment:
if ``complevel`` is not None and ``complib`` is None then you can set the default to zlib, otherwise the *user* must specific a compression to make this active.